api/v1/requests/AttendeeRequest.js

```diff
@@ -0,0 +1,65 @@
+var Request = require('./Request');
+var validators = require('../utils/validators');
+var registration = require('../utils/registration');
+
+var extraInfoValidations = {
+	info:       ['string', 'maxLength:255']
+};
+
+var projectValidations = {
+	name:       ['required', 'string', 'maxLength:100'],
+	description:['required', 'string', 'maxLength:255'],
+	repo:       ['required', 'string', 'maxLength:150'],
+	isSuggestion: ['required', 'boolean']
+};
+
+var ecosystemInterestValidations = {
+	ecosystemId:  ['required', 'integer']
+};
+
+var requestedCollaboratorValidations = {
+	collaborator: ['required', 'string', 'maxLength:255']
+};
+
+var bodyRequired = ['attendee', 'ecosystemInterests'];
+var bodyAllowed = ['projects', 'extras', 'collaborators'];
+var bodyValidations = {
+	'attendee': ['required', 'plainObject'],
+	'attendee.firstName': ['required', 'string', 'maxLength:255'],
+	'attendee.lastName': ['required', 'string', 'maxLength:255'],
+	'attendee.shirtSize': ['required', 'string', registration.verifyTshirtSize],
+	'attendee.diet': ['required', 'string', registration.verifyDiet],
+	'attendee.age': ['required', 'integer', 'min:13', 'max:115'],
+	'attendee.transportation': ['required', 'string', registration.verifyTransportation],
+	'attendee.school': ['required', 'string', 'maxLength:255'],
+	'attendee.major': ['required', 'string', 'maxLength:255'],
+	'attendee.gender': ['required', 'string', registration.verifyGender],
+	'attendee.professionalInterest': ['required', 'string', registration.verifyProfessionalInterest],
+	'attendee.github': ['required', 'string', 'maxLength:50'],
+	'attendee.interests': ['required', 'string', 'maxLength:255'],
+	'attendee.status': ['string', registration.verifyStatus],
+	'attendee.isNovice': ['required', 'boolean'],
+	'attendee.isPrivate': ['required', 'boolean'],
+	'ecosystemInterests': ['required', 'array', 'minLength:1', 'maxLength:2', validators.array(validators.nested(ecosystemInterestValidations, 'ecosystemInterests'), 'ecosystemInterests')],
+	'projects': ['array', 'minLength:1', 'maxLength:2', registration.verifyProjectArray, validators.array(validators.nested(projectValidations, 'projects'), 'projects')],
```

Comment:
On a side-note, the `projects` should be completely optional, not `minLength:1`. I'll change that.